scripts/us_census/pep/monthly_population_estimate/preprocess.py

```diff
@@ -0,0 +1,369 @@
+
+        Arguments:
+            file (str) : String of Dataset File Path
+        Returns:
+            df (DataFrame) : DataFrame with loaded dataset
+        """
+        df = None
+        self.file_name = os.path.basename(file)
+        if ".xls" in file:
+            df = pd.read_excel(file)
+        return df
+
+    def _transform_df(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        This method transforms Dataframe into cleaned DF.
+        Also, It Creates new columns, remove duplicates,
+        Standaradize headers to SV's, Mulitply with
+        scaling factor.
+
+        Arguments:
+            df (DataFrame) : DataFrame
+
+        Returns:
+            df (DataFrame) : DataFrame.
+        """
+        final_cols = [col for col in df.columns if 'year' not in col.lower()]
+
+        df['Year'] = df['Year and Month'].apply(_return_year).fillna(
+            method='ffill', limit=12)
+        df['Month'] = df['Year and Month'].apply(_return_month)
+        df.dropna(subset=['Year', 'Month'], inplace=True)
+
+        # Creating new Date Column and Final Date format is yyyy-mm
+        df['Date'] = df['Year'] + df['Month']
+        df['Date'] = df['Date'].str.replace(".", "").str.replace(
+            " ", "").str.replace("*", "")
+        df['Date'] = pd.to_datetime(df['Date'],
+                                    format='%Y%B%d',
+                                    errors="coerce")
+        df.dropna(subset=['Date'], inplace=True)
+        df['Date'] = df['Date'].dt.strftime('%Y-%m')
+        df.drop_duplicates(subset=['Date'], inplace=True)
+
+        # Deriving new SV Count_Person_InArmedForcesOverseas as
+        # subtracting Resident Population from
+        # Resident Population Plus Armed Forces Overseas
+        df['Count_Person_InUSArmedForcesOverseas'] = df[
+            'Resident Population Plus Armed Forces Overseas'].astype(
+                'int') - df['Resident Population'].astype('int')
+        computed_cols = ["Date", "Count_Person_InUSArmedForcesOverseas"]
+
+        # Selecting Coumputed and Final Columns from the DF.
+        df = df[computed_cols + final_cols]
+
+        # Renaming DF Headers with ref to SV's Naming Standards.
+        final_cols_list = ["Count_Person_" + col\
+                        .replace("Population ", "")\
+                        .replace("Population", "")\
+                        .replace(" Plus ", "Or")\
+                        .replace("Armed Forces Overseas", \
+                            "InUSArmedForcesOverseas")\
+                        .replace("Household", \
+                                "ResidesInHousehold")\
+                        .replace("Resident", "USResident")\
+                        .replace("Noninstitutionalized", \
+                            "NonInstitutionalized")\
+                        .strip()\
+                        .replace(" ", "_")\
+                        .replace("__", "_")\

```

Comment:
Can you use a helper function for this?

Reply:
Created a new helper function to replace the strings